pkg/cache/azuresa_based.go

```diff
@@ -20,6 +20,48 @@ type AzureCache struct {
 	session       *azblob.Client
 }
 
+type AzureCacheConfiguration struct {
+	StorageAccount string `mapstructure:"storageaccount" yaml:"storageaccount,omitempty"`
+	ContainerName  string `mapstructure:"container" yaml:"container,omitempty"`
+}
+
+func (s *AzureCache) Configure(cacheInfo CacheProvider) error {
+	s.ctx = context.Background()
+	if cacheInfo.Azure.ContainerName == "" {
+		log.Fatal("Azure Container name not configured")
+	}
+	if cacheInfo.Azure.StorageAccount == "" {
+		log.Fatal("Azure Storage account not configured")
+	}
+
+	// We assume that Storage account is already in place
+	blobUrl := fmt.Sprintf("https://%s.blob.core.windows.net/", cacheInfo.Azure.StorageAccount)
+	credential, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	client, err := azblob.NewClient(blobUrl, credential, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Try to create the blob container
+	_, err = client.CreateContainer(s.ctx, cacheInfo.Azure.ContainerName, nil)
+	if err != nil {
+		// TODO: Maybe there is a better way to check this?
+		// docs: https://pkg.go.dev/github.com/Azure/azure-storage-blob-go/azblob
+		if strings.Contains(err.Error(), "ContainerAlreadyExists") {
+			// do nothing
+		} else {
+			log.Fatal(err)
```

Comment:
If this configure is being called whilst in server mode, won't this crash the server out?

Reply:
yeah I had another look, I suggest to just return the error upon a failed blob storage/gcs bucket creation and also return an error on the s3 equivalent code
https://github.com/k8sgpt-ai/k8sgpt/pull/750/files#diff-428e509fd6888552bb5dddfd3032c43c016def1c99fdf9658f6b65419bd92c79

Reply:
A new commit is available with the modification.